MobilityData/mobility-feed-api
# fix: UI OpenApi generator fails to create a proper Feed's discriminator

# Summary:

Fixes: #560 
The TS types are included as part of the repository. We are using the generate:api-types library to generate the types. Until now, it was at the developers' discretion to run the generator. A breaking change was introduced in a [previous PR](https://github.com/MobilityData/mobility-feed-api/pull/394), and it was not until last week we realized that the TS types were broken when trying to generate the new version. The breaking changes were necessary because the Python fastApi generator, used on the server side, [doesn't fully support the `anyOf` or `oneOf` OpenApi feature](https://openapi-generator.tech/docs/generators/python-fastapi/#schema-support-feature). 

## Changes:

- Replace `allOf` by a schema component containing all properties from BasicFeed, GtfsFeed and GtfsRTFeed.
- Add a `discriminator` to the BasicFeed to support TS code generation.
- Added a Github action for PRs that changed the OpenApi schema if the TS script generator was executed and it contains the expected types, [example failing action](https://github.com/MobilityData/mobility-feed-api/actions/runs/9947302189/job/27479691559)

## Testing tips:

- Execute the smoke test on the UI, ensuring the Home, Feeds, and Feed pages render the correct information.
- Start the API locally and verify that the search endpoint returns the same JSON structure as before(QA).

Please make sure these boxes are checked before submitting your pull request - thanks!
